src/rsa/signing.rs

```diff
@@ -176,10 +213,18 @@ impl RSAKeyPair {
                     return Err(error::Unspecified);
                 }
 
+                // [NIST SP-800-56B rev. 1]] 6.4.1.4.3 - Step 6
+                //
+                // TODO 6.a: d < LCM(p – 1, q – 1)
+                // TODO 6.b: 1 == (d · e) mod LCM(p – 1, q – 1)
```

Comment:
We don't use `d`, but we do use `dP` and `dQ`. There's already a TODO in the code about validating `dP` and `dQ`, (predating this PR), which relies on the same argument about checking `(b<sup>d</sup>)<sup>e</sup> == b (mod n)` during signing. Meanwhile, we do validate up front that `pq = n`. 

So, perhaps the documentation should say instead:

* While some size checks of `d`, `dP` and `dQ` are performed at construction, the validity of the mathematical relationship of `dP`, `dQ`, `e` and `n` is verified only during signing, when we validate that `(b**d)**e == b mod n` (having calculated `b**d` using `dP` and `dQ`). If a preemptive check of the consistency of these values with each other is necessary then that can be done by signing any message with the key pair.
* `d` is fully validated neither at construction nor during signing. This is OK as far as _ring_'s usage of the key is concerned because _ring_ never uses the value of `d` (_ring_ always uses `p`, `q`, `dP` and `dQ` via the Chinese Remainder Theorem, instead of using `d`). However, _ring_'s checks would not be sufficient for validating a key pair for use by some other system; that other system must check the value of `d` itself if it uses it.

WDYT?